wp/lib/bap_wp/src/output.ml

```diff
@@ -200,11 +200,15 @@ let output_gdb (solver : Solver.solver) (status : Solver.status)
     let model = Constr.get_model_exn solver in
     let mem_model = get_mem model env in
     let varmap = Env.get_var_map env in
-    let module Target = (val target_of_arch (Env.get_arch env)) in
-    let regmap = VarMap.filter_keys ~f:(Target.CPU.is_reg) varmap in
+    let target = (Env.get_arch env) in
+    let to_core v = Theory.Var.create (Var.sort v) (Var.ident v) in
+    (* FIXME: does this return [mem] by accident? *)
```

Comment:
Can you clarify this section? Is `to_core` the one returning `mem`, and is `is_reg` properly filtering it out?

Reply:
`to_core` is just a type cast from `Bap.Std.Var.t` to `Theory.Var.t`.

I'll clarify what this code is doing in comments.